iommu_intro.adoc

```diff
@@ -573,60 +585,63 @@ access privileges. As the IOMMU itself is a bus master for its implicit
 accesses, the IOMMU hardware ID may be used by the PMP to select the appropriate
 access control rules.
 
+[NOTE]
+====
+The IOMMU does not validate the authenticity of the hardware IDs provided by 
+the IO bridge. 
+
+The IO bridge and/or the root ports must include suitable mechanisms to
+authenticate the hardware IDs. In some SOC this may be trivially achieved as a
+property of the devices being integrated into the SOC and their IDs being
+immutable. For PCIe, for example, the PCIe defined Access Control Services (ACS)
+Source Validation capabilities may be used to authenticate the hardware IDs.
+Other implementation specific methods in the IO bridge may be provided to
+perform such authentication.
+====
+
 === IOMMU features
 The version 1.0 of the RISC-V IOMMU specification supports the following
 features:
 
 * Memory-based device context to locate parameters and address translations
   structures. The device context is located using the hardware provided
   unique `device_id`. The supported `device_id` width may be up to 24-bit.
-  IOMMU is required to support at least one of the valid `device_id` widths as
-  specified in <<DATA_STRUCTURES>>.
 
 * Memory-based process context to locate parameters and address translation
   structures using hardware provide unique `process_id`. The supported
-  `process_id` may be up to 20-bit. IOMMU is required to support at least one
-  of the valid `process_id` widths as specified in <<DATA_STRUCTURES>>
+  `process_id` may be up to 20-bit.
 
-* IOMMU must support 16-bit GSCIDs and 20-bit PSCIDs.
+* 16-bit GSCIDs and 20-bit PSCIDs.
 
-* An implementation may support only the VS/S-stage of address translation,
-  only G-stage address translation, or two stage address translation.
+* Single stage and two stage address translation.
 
-* VS/S-stage and/or G-stage virtual-memory system as specified by the RISC-V
+* VS/S-stage and G-stage virtual-memory system as specified by the RISC-V
   privileged specification to allow software flexibility to use a common page
-  table for the CPU MMU as well as the IOMMU or to use a separate page table for the
-  IOMMU.
+  table for the CPU MMU as well as the IOMMU or to use a separate page table
+  for the IOMMU.
 
-* Up to 57-bit virtual-address width and 59-bit guest-physical-address width.
+* Up to 57-bit virtual-address width, 56-bit system-physical-address, and 59-bit
+  guest-physical-address width.
 
-* Support for hardware management of page-table entry Accessed and Dirty bits
-  is optional for the IOMMU.
+* Hardware updating of PTE Accessed and Dirty bits.
 
-* Support for MSI address translation to redirect MSIs to interrupt files in
-  an IMSIC is optional. When MSI address translation is supported using flat
-  MSI page tables then supporting memory-resident-interrupt-files is optional.
+* MSI address translation to redirect MSIs to interrupt files in an IMSIC 
```

Comment:
MSI address translation to redirect MSIs to interrupt files in an IMSIC is supported even if the MSI_FLAT is 0. It uses G-stage only translation to perform the redirection.

Reply:
I will update to state "using MSI page tables"